File: apps/web/test_browser.py
from playwright.sync_api import sync_playwright

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()
        
        page.on("console", lambda msg: print(f"Browser Console: {msg.text}"))
        page.on("pageerror", lambda err: print(f"Browser Error: {err}"))
        
        print("Navigating to http://localhost:3000/history")
        page.goto("http://localhost:3000/history", wait_until="networkidle")
        print("Page loaded.")
        
        browser.close()

run()

File: apps/web/test-browser.js
const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  
  page.on('console', msg => console.log('PAGE LOG:', msg.text()));
  page.on('pageerror', error => console.log('PAGE ERROR:', error.message));
  
  await page.goto('http://localhost:3000/resume-builder', { waitUntil: 'networkidle0' });
  
  // Wait a bit to see if React catches an error and prints it
  await new Promise(r => setTimeout(r, 2000));
  
  await browser.close();
})();
